src/test/java/org/gridsuite/study/server/NonEvacuatedEnergyTest.java

```diff
@@ -0,0 +1,728 @@
+        NetworkModificationNode modificationNode1 = createNetworkModificationNode(studyUuid, rootNodeUuid, UUID.randomUUID(), VARIANT_ID, "node 1");
+        UUID modificationNodeUuid = modificationNode1.getId();
+        networkModificationTreeService.updateNonEvacuatedEnergyResultUuid(modificationNodeUuid, notFoundSensitivityUuid);
+        assertTrue(networkModificationTreeService.getNonEvacuatedEnergyResultUuid(modificationNodeUuid).isPresent());
+        assertEquals(notFoundSensitivityUuid, networkModificationTreeService.getNonEvacuatedEnergyResultUuid(modificationNodeUuid).get());
+
+        wireMock.stubFor(WireMock.get(WireMock.urlPathMatching("/v1/non-evacuated-energy-results/" + notFoundSensitivityUuid))
+                .willReturn(WireMock.notFound()));
+
+        mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/non-evacuated-energy/result", studyUuid, modificationNodeUuid, FAKE_NON_EVACUATED_ENERGY_RESULT_JSON)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound()).andReturn();
+    }
+
+    @Test
+    @SneakyThrows
+    public void testResetUuidResultWhenNonEvacuatedEnergyFailed() {
+        UUID resultUuid = UUID.randomUUID();
+        StudyEntity studyEntity = insertDummyStudy(UUID.randomUUID(), UUID.randomUUID());
+        RootNode rootNode = networkModificationTreeService.getStudyTree(studyEntity.getId());
+        NetworkModificationNode modificationNode = createNetworkModificationNode(studyEntity.getId(), rootNode.getId(), UUID.randomUUID(), VARIANT_ID, "node 1");
+        String resultUuidJson = mapper.writeValueAsString(new NodeReceiver(modificationNode.getId()));
+
+        // Set an uuid result in the database
+        networkModificationTreeService.updateNonEvacuatedEnergyResultUuid(modificationNode.getId(), resultUuid);
+        assertTrue(networkModificationTreeService.getNonEvacuatedEnergyResultUuid(modificationNode.getId()).isPresent());
+        assertEquals(resultUuid, networkModificationTreeService.getNonEvacuatedEnergyResultUuid(modificationNode.getId()).get());
+
+        StudyService studyService = Mockito.mock(StudyService.class);
+        doAnswer(invocation -> {
+            input.send(MessageBuilder.withPayload("").setHeader(HEADER_RECEIVER, resultUuidJson).build(), nonEvacuatedEnergyFailedDestination);
+            return resultUuid;
+        }).when(studyService).runNonEvacuatedEnergy(any(), any(), any());
+        studyService.runNonEvacuatedEnergy(studyEntity.getId(), modificationNode.getId(), "userId");
+
+        Message<byte[]> message = output.receive(TIMEOUT, studyUpdateDestination);
+        assertEquals(studyEntity.getId(), message.getHeaders().get(NotificationService.HEADER_STUDY_UUID));
+        String updateType = (String) message.getHeaders().get(NotificationService.HEADER_UPDATE_TYPE);
+        assertEquals(NotificationService.UPDATE_TYPE_NON_EVACUATED_ENERGY_FAILED, updateType);
```

Comment:
we don't check that resultUuid is cleared ?

Reply:
Check added